addons/stock_account/models/product.py

```diff
@@ -491,9 +491,11 @@ def _run_fifo_vacuum(self, company=None):
         # If some negative stock were fixed, we need to recompute the standard price.
         for product in self:
             product = product.with_company(company.id)
-            if product.cost_method == 'average' and not float_is_zero(product.quantity_svl,
-                                                                      precision_rounding=product.uom_id.rounding):
-                product.sudo().with_context(disable_auto_svl=True).write({'standard_price': product.value_svl / product.quantity_svl})
+            # Only recompute if we fixed some negative stock
+            if svls_to_vacuum_by_product[product.id]:
```

Comment:
```suggestion
            if not svls_to_vacuum_by_product[product.id]:
                continue
```
this way there is no need to change the other lines